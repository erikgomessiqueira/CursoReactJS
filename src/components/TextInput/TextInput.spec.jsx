import { render, screen } from '@testing-library/react';
import { TextInput } from '.';
import userEvent from '@testing-library/user-event';

describe('<TextInput/>', () => {
  it('should have a value of searchValue', () => {
    const fn = jest.fn();
    render(<TextInput handleChange={fn} searchValue="Teste" />);

    const input = screen.getByPlaceholderText(/Pesquisa/i);
    expect(input).toHaveAttribute('value', 'Teste');
  });

  it('should call handleChange function on each key pressed', () => {
    const fn = jest.fn();
    render(<TextInput handleChange={fn} searchValue="Valor" />);

    const input = screen.getByPlaceholderText(/Pesquisa/i);

    const testValue = 'Testando fn()';
    userEvent.type(input, testValue);

    expect(input.value).toBe('Valor');
    expect(fn).toHaveBeenCalledTimes(testValue.length);
  });

  it('should match sanpshot', () => {
    const fn = jest.fn();
    const { container } = render(<TextInput handleChange={fn} searchValue="Teste" />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
