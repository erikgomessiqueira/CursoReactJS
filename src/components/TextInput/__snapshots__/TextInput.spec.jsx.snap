// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextInput/> should match sanpshot 1`] = `
<input
  class="text-input"
  placeholder="Pesquisa"
  type="search"
  value="Teste"
/>
`;
